src/components/departure-date-selector/index.tsx

```diff
@@ -0,0 +1,114 @@
+import { ChangeEvent, useState } from 'react';
+import style from './departure-date-selector.module.css';
+import { ComponentText, Language, useTranslation } from '@atb/translations';
+import { TFunc } from '@leile/lobo-t';
+
+export enum DepartureDateState {
+  Now = 'now',
+  Arrival = 'arrival',
+  Departure = 'departure',
+}
+
+type DepartureDateSelectorProps = {
+  initialState?: DepartureDateState;
+  onStateChange: (state: DepartureDateState) => void;
+  onDateChange: (date: Date) => void;
+  onTimeChange: (time: string) => void;
+};
+
+export default function DepartureDateSelector({
+  initialState = DepartureDateState.Now,
+  onStateChange,
+  onDateChange,
+  onTimeChange,
+}: DepartureDateSelectorProps) {
+  const { t } = useTranslation();
+  const [selectedOption, setSelectedOption] =
+    useState<DepartureDateState>(initialState);
+  const [selectedDate, setSelectedDate] = useState(new Date());
+  const [selectedTime, setSelectedTime] = useState(() => {
+    const now = new Date();
+    const hours = now.getHours().toString().padStart(2, '0');
+    const minutes = now.getMinutes().toString().padStart(2, '0');
+
+    return `${hours}:${minutes}`;
+  });
+
+  const internalOnStateChange = (state: DepartureDateState) => {
+    setSelectedOption(state);
+    onStateChange(state);
+  };
+  const internalOnDateChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setSelectedDate(new Date(event.target.value));
+    onDateChange(new Date(event.target.value));
+  };
+
+  const internalOnTimeChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setSelectedTime(event.target.value);
+    onTimeChange(event.target.value);
+  };
+
+  return (
+    <div className={style.departureDateSelector}>
+      <div className={style.options}>
+        {Object.values(DepartureDateState).map((state) => (
+          <div key={state} className={style.option}>
+            <label>
+              <input
+                type="radio"
+                name="departureDateSelector"
+                value={state}
+                checked={selectedOption === state}
+                onChange={() => internalOnStateChange(state)}
+                aria-label={stateToLabel(state, t)}
+              />
+              <span aria-hidden>{stateToLabel(state, t)}</span>
+            </label>
+          </div>
+        ))}
+      </div>
+
+      {selectedOption !== DepartureDateState.Now && (
+        <div className={style.dateAndTimeSelectors}>
+          <div className={style.dateSelector}>
+            <label htmlFor="departureDateSelector">
+              {t(ComponentText.DepartureDateSelector.date)}
+            </label>
+            <input
```

Comment:
Looks like these inputs have `color: var(--text-colors-secondary);` in the Figma sketches